### ISSUE TITLE
Final Fantasy Ever Crisis

### ISSUE THREAD

**Issue by Asciicodeplus** [190nm/rein-kuro#6]
Looks like this game uses the same file system, it has a bunch of resources with 32 character names and an octocacheevai file, I tried to decrypt the files with this tool but I get the following error:

English
>>> [Error] Key b'D`\x9f\x8cFc\x142J^s\xdc\xae\xc8nx'is incorrect. <<<

Japanese
>>> [Error] Key b'\x9e\xfc\x98\x8c\xccI\x9aD\xbb\x14\xee\x92)\x988w'is incorrect. <<<

I guess the tool has a hardcoded key for each version of the Nier game, can you take a look at FF7EC? here are a few file samples:

https://mega.nz/folder/wD5xgDCI#KnXccKwoPVsMphZM5o7JWQ

**Comment by 09williamsad** [comment]
Not the same system.
See https://github.com/FF7-Ever-Crisis/Resources

**Comment by 190nm** [comment]
> Looks like this game uses the same file system, it has a bunch of resources with 32 character names and an octocacheevai file, I tried to decrypt the files with this tool but I get the following error:
> 
> English
> 
> > > > [Error] Key b'D`\x9f\x8cFc\x142J^s\xdc\xae\xc8nx'is incorrect. <<<
> 
> Japanese
> 
> > > > [Error] Key b'\x9e\xfc\x98\x8c\xccI\x9aD\xbb\x14\xee\x92)\x988w'is incorrect. <<<
> 
> I guess the tool has a hardcoded key for each version of the Nier game, can you take a look at FF7EC? here are a few file samples:
> 
> https://mega.nz/folder/wD5xgDCI#KnXccKwoPVsMphZM5o7JWQ

Yes that's correct, the tool did have a hard coded key specific to the game. If the project linked in the comment above me doesn't help you, here's what I remember about getting the key manually: 

It was buried in the metadata of an unassuming png file included in the app. IIRC it was one of the low res app icons. An obfuscated function with a name like "GetImageSizeAndData" or so, was being called during decryption, and that's how I found where it was hidden.

Hope this helps

**Comment by Asciicodeplus** [comment]
Seems like someone figured out the key for the mobile english version of the game but I havent been able to make it work, I'm guessing I need a matching "iv" too:

il2cpp: 0x04c06a70 ┌─Octo.Data.AESCrypt..ctor(this = Octo.Data.AESCrypt, key = "57lk6xzzeqmy4qay", mode = CBC, padding = PKCS7);
il2cpp: 0x04c06b00 │ ┌─Octo.Data.AESCrypt..ctor(this = Octo.Data.AESCrypt, key = [49,225,112,221,128,118,1,143,19,40,39,17,133,139,123,37], mode = CBC, padding = PKCS7);
il2cpp: 0x04c06b00 │ └─Octo.Data.AESCrypt..ctor;

This is the error I get:

Namespace(decrypt=all, localization=en, revision=None, key='35376C6B36787A7A65716D7934716179', iv=None)

key: 60a50f548996e9f8a86ea85317a90a74
iv : 1c6e6f9255c0e5412712f4010225e378

>>> [Error] Key b'`\xa5\x0fT\x89\x96\xe9\xf8\xa8n\xa8S\x17\xa9\nt'is incorrect. <<<
